src/ra_fwd.h

```diff
@@ -2,34 +2,124 @@
 
+#define _NORETURN [[noreturn]]
 
-namespace ra {
+#if _HAS_CXX17
+#define _DEPRECATED          [[deprecated]]
+#define _DEPRECATEDR(reason) [[deprecated(reason)]]
+#define _FALLTHROUGH         [[fallthrough]] /* ; you need ';' at the end */
+#define _UNUSED              [[maybe_unused]]
+#define _CONSTANT_VAR        inline constexpr auto
+#else
+#define _NODISCARD           _Check_return_
+#define _DEPRECATED          __declspec(deprecated)
+#define _DEPRECATEDR(reason) _CRT_DEPRECATE_TEXT(reason)
+#define _FALLTHROUGH         __fallthrough /* ; you need ';' at the end */
+#define _UNUSED              
+#define _CONSTANT_VAR        constexpr auto
+#endif /* _HAS_CXX17 */        
+
+#define _CONSTANT_LOC constexpr auto /* local vars can't be inline */
+#define _CONSTANT_FN  _CONSTANT_VAR
+
+#ifndef CALLBACK
+#define CALLBACK __stdcall
+#endif /* !CALLBACK */
 
 #ifndef _TCHAR_DEFINED
 #if _MBCS
-using TCHAR = char;
+using TCHAR  = char;
+using PTCHAR = char*;
+using TBYTE  = unsigned char;
+using PTBYTE = unsigned char*;
 #elif _UNICODE
-using TCHAR = wchar_t;
+using TCHAR  = wchar_t;
+using PTCHAR = wchar_t*;
+using TBYTE  = wchar_t;
+using PTBYTE = wchar_t*;
 #else 
 #error Unknown character set detected, only MultiByte and Unicode are supported!
-#endif // _MBCS
+#endif /* _MBCS */
 #define _TCHAR_DEFINED
-#endif // !_TCHAR_DEFINED
+#endif /* !_TCHAR_DEFINED */
+
+#ifndef _WINNT_
+using HANDLE  = void*;
+using LPCTSTR = const TCHAR*;
+using LPTSTR  = TCHAR*;
+#endif /* !_WINNT_ */
+
+#ifndef _BASETSD_H_
+#if _WIN32
+using LONG_PTR = long;
+#elif _WIN64
+using LONG_PTR = long long;
+#else
+#error Only Windows is currently supported
+#endif /* _WIN32 */
+#endif /* !_BASETSD_H_ */
+
+#ifndef _WINDEF_
+using BOOL   = int;
+using BYTE   = unsigned char;
+using DWORD  = unsigned long;
+using LPVOID = void*;
+
+struct HBITMAP__;
+struct HDC__;
+struct HPEN__;
+struct HFONT__;
+struct HBRUSH__;
+struct HINSTANCE__;
+struct HWND__;
+struct tagRECT;
 
+using HBITMAP   = HBITMAP__*;
+using HDC       = HDC__*;
+using HPEN      = HPEN__*;
+using HFONT     = HFONT__*;
+using HBRUSH    = HBRUSH__*;
+using HINSTANCE = HINSTANCE__*;
+using HMODULE   = HINSTANCE;
+using HWND      = HWND__*;
+using RECT      = tagRECT;
 
+using UINT    = unsigned int;
+using LPARAM  = LONG_PTR;
+using LRESULT = LONG_PTR;
+using WPARAM  = std::uintptr_t;
+using INT_PTR = std::intptr_t;
+#endif /* !_WINDEF_ */
 
-using tstring = std::basic_string<TCHAR>;
+/* This needs to be forward declared to get rid of a conformance error */
+struct IUnknown;
+
+namespace std {
+
+using ::time_t;
+
+} /* namespace std */
+
+namespace ra {
+
+/* usings/typedefs matter in C++ */
+#if _MBCS
+using tstring = std::basic_string<char, std::char_traits<char>, std::allocator<char>>;
+#elif  _UNICODE
+using tstring = std::basic_string<wchar_t, std::char_traits<wchar_t>, std::allocator<wchar_t>>;
+#else
+#error Unknown Character set
+#endif /* _MBCS */
```

Comment:
Was attempting to make this file "include-less" but it was too much work and would a lot more lines added than already present so I just wanted to save it for later.